### PR TITLE
Make JetStream the default backend on prod

### PR DESCRIPTION
**Pull request by raypinto**
**Description**

Changes proposed in this pull request:

- Enable JetStream by default

**Related issue(s)**

### CODE REVIEW

File: docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md [hunk 1]
```diff
@@ -12,30 +12,31 @@ Events were not received by the consumers.
...
-       ```
-       
-    To correlate the consumer to the Subscription and the specific event type, check the `description` field of the consumer.
+       4. Check if the consumers were created and have the expected configurations.
+          ```bash
+          nats consumer info
+          ```
+          To correlate the consumer to the Subscription and the specific event type, check the `description` field of the consumer.
+
+       5. If the PVC storage is fully consumed, and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which will remove the old messages.
```
Comment (VladislavPaskar) [inline docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md:39]:
```suggestion
       5. If the PVC storage is fully consumed and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which will remove the old messages.
```
```suggestion
       5. If the PVC storage is fully consumed, and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which will remove the old messages.
```

Comment (VladislavPaskar) [inline docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md:39]:
```suggestion
       5. If the PVC storage is fully consumed and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which will remove the old messages.
```
```suggestion
       5. If the PVC storage is fully consumed, and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which will remove the old messages.
```

File: docs/05-technical-reference/00-architecture/evnt-01-architecture.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Eventing Controller manages the internal infrastructure in order to receive an e
 
 ## JetStream
 
-Kyma now supports JetStream, which is a persistence offering from NATS, that guarantees `at least once` delivery. It is built-in within our default NATS backend. Follow the [tutorial](../../04-operation-guides/operations/evnt-01-enable-kyma-with-jetstream.md) to enable Kyma with JetStream.
+Kyma now supports JetStream by default, which is a persistence offering from NATS, that guarantees `at least once` delivery. It is built-in within our default NATS backend.
```
Comment (VladislavPaskar) [inline docs/05-technical-reference/00-architecture/evnt-01-architecture.md:37]: (nit)
I am not sure it's the best to formulate this. Perhaps something like: `JetStream NATS is enabled in Kyma by default, which is...`

Comment (VladislavPaskar) [inline docs/05-technical-reference/00-architecture/evnt-01-architecture.md:37]: (nit)
I am not sure it's the best to formulate this. Perhaps something like: `JetStream NATS is enabled in Kyma by default, which is...`

File: tests/fast-integration/eventing-test/eventing-test.js [hunk 3]
```diff
@@ -124,6 +128,12 @@ describe('Eventing tests', function() {
         eventingSubscription(`sap.kyma.custom.commerce.order.created.v1`, sink, 'order-created', testNamespace),
       ];
 
+      before('check if at least once delivery tests need to be skipped', async function() {
```
Comment (VladislavPaskar) [inline tests/fast-integration/eventing-test/eventing-test.js:131]: (nit)
I would better log, when skip the test and not the check. Something like:

```
 before('', async function() {
        if (skipAtLeastOnceDeliveryTest()) {
          console.log('Skipping the at least once delivery tests for non-limits retention policy');
          this.skip();
        }
      });
```

Comment (VladislavPaskar) [inline tests/fast-integration/eventing-test/eventing-test.js:131]: (nit)
I would better log, when skip the test and not the check. Something like:

```
 before('', async function() {
        if (skipAtLeastOnceDeliveryTest()) {
          console.log('Skipping the at least once delivery tests for non-limits retention policy');
          this.skip();
        }
      });
```

File: tests/fast-integration/eventing-test/utils.js [hunk 4]
```diff
@@ -90,6 +89,31 @@ async function getNatsPods() {
   return await listPods(labelSelector, 'kyma-system');
 }
 
+async function getStreamConfigForJetStream() {
```
Comment (VladislavPaskar) [inline tests/fast-integration/eventing-test/utils.js:92]: (nit)
could you add a description to this function ?

Comment (VladislavPaskar) [inline tests/fast-integration/eventing-test/utils.js:92]: (nit)
could you add a description to this function ?

**Review by VladislavPaskar — DISMISSED** [review]

**Review by VladislavPaskar — DISMISSED** [review]

File: docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md [hunk 5]
```diff
@@ -12,30 +12,31 @@ Events were not received by the consumers.
...
-        │                                  Streams                                   │
-        ├──────┬─────────────┬─────────────────────┬──────────┬───────┬──────────────┤
-        │ Name │ Description │ Created             │ Messages │ Size  │ Last Message │
-        ├──────┼─────────────┼─────────────────────┼──────────┼───────┼──────────────┤
-        │ sap  │             │ 2022-05-03 00:00:00 │ 0        │ 318 B │ 5.80s        │
-        ╰──────┴─────────────┴─────────────────────┴──────────┴───────┴──────────────╯
-        ```       
-
-    3. If the stream exists, check the timestamp of the `Last Message` that the stream received. A recent timestamp would mean that the event was published correctly.
+       1. Port forward the Kyma Eventing NATS Service to localhost. Use port `4222`. Run:
```
Comment (grego952) [inline docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md:15]: For lines 15-31, make sure that only code is in the code block. The text should be outside of it.

File: tests/fast-integration/eventing-test/README.md [hunk 6]
```diff
@@ -51,6 +49,7 @@ npm run test-eventing
 ```bash
 npm run test-eventing TEST_SUFFIX=abcd
 ```
+>**NOTE:** The `at least once` delivery test for JetStream will only be run when the `STREAM_RETENTION_POLICY` is set to `limits` and the `CONSUMER_DELIVER_POLICY` is set to `all`.
```
Comment (grego952) [inline tests/fast-integration/eventing-test/README.md:52]:
```suggestion
>**NOTE:** The `at least once` delivery test for JetStream is only run when the `STREAM_RETENTION_POLICY` is set to `limits` and the `CONSUMER_DELIVER_POLICY` is set to `all`.
```

File: docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md [hunk 7]
```diff
@@ -12,30 +12,31 @@ Events were not received by the consumers.
...
-       ```
-       
-    To correlate the consumer to the Subscription and the specific event type, check the `description` field of the consumer.
+       4. Check if the consumers were created and have the expected configurations.
+          ```bash
+          nats consumer info
+          ```
+          To correlate the consumer to the Subscription and the specific event type, check the `description` field of the consumer.
+
+       5. If the PVC storage is fully consumed and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which will remove the old messages.
```
Comment (grego952) [inline docs/04-operation-guides/troubleshooting/evnt-02-jetstream-troubleshooting.md:39]:
```suggestion
       5. If the PVC storage is fully consumed and matches the stream size as shown above, the stream can no longer receive messages. Either increase the PVC storage size or set the `MaxBytes` property which removes the old messages.
```

**Review by grego952 — DISMISSED** [review]

**Review by VladislavPaskar — DISMISSED** [review]

**Review by VladislavPaskar — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**